### PR TITLE
Add swift+linux deprecation

### PR DESCRIPTION
**Pull request by aeisenberg**
### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,15 +6,15 @@ Note that the only difference between `v2` and `v3` of the CodeQL Action is the
 
 ## [UNRELEASED]
 
-No user facing changes.
+- _Deprecation:_ Analyzing databases on Swift support on Ubuntu runner images is no longer supported. Please migrate to a MacOS runner if this affects you.
```
Comment (adityasharad) [inline CHANGELOG.md:9]:
```suggestion
- _Deprecation:_ Swift analysis on Ubuntu runner images is no longer supported. Please migrate to a macOS runner if this affects you.
```

Comment (adityasharad) [inline CHANGELOG.md:9]: Can we get the Action to produce a deprecation warning if it detects the combination of Swift and Linux?

Comment (aeisenberg) [reply CHANGELOG.md:9]: Yes. @angelapwen will work on that.

Comment (aeisenberg) [reply CHANGELOG.md:9]: Hmm...my sentence didn't even make sense. 😨 This is much better.

**Review by adityasharad — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,15 +6,15 @@ Note that the only difference between `v2` and `v3` of the CodeQL Action is the
 
 ## [UNRELEASED]
 
-No user facing changes.
+- _Deprecation:_ Swift analysis on Ubuntu runner images is no longer supported. Please migrate to a macOS runner if this affects you.
```
Comment (angelapwen) [inline CHANGELOG.md:9]: Do we want a link to this PR for consistency?

File: CHANGELOG.md (hunk 1 above)
Comment (angelapwen) [reply CHANGELOG.md]: I've added the warning in #2399.

File: CHANGELOG.md (hunk 2 above)
Comment (aeisenberg) [reply CHANGELOG.md:9]: Now that we have a PR, I will add it.

**Review by angelapwen — APPROVED** [review]

**Pull request merged**